concepts/admin.py

```diff
@@ -8,16 +11,46 @@ def resolve(modeladmin, request, queryset):
         authorities.resolve(type(obj), obj)
 resolve.verbose_name = 'resolve selected concepts'
 
+def merge_concepts(modeladmin, request, queryset):
+
+    opts = modeladmin.model._meta
+    app_label = opts.app_label
+    RESOLVED_STATUS = 'Resolved'
+    resolvedCount = 0
+    resolvedConcept = None
+
+    for item in queryset:
```

Comment:
Rather than iterate over the entire queryset (which loads the whole result set from the database), we can cut down on our DB overhead by using queryset methods. E.g.

```python
resolvedCount = queryset.filter(concept_state=Concept.RESOLVED).count()
```